### ISSUE TITLE
An alternative to Session Buddy: "Tabs Backup & Restore"

### ISSUE THREAD

**Issue by mkrier** [deanoemcke/thegreatsuspender#742]
When the recent Suspender update screwed my tabs, I thought, no problem, I've got Session Buddy, as many have mentioned here.  *sigh*.  No. 

Well, guess what- apparently, a year ago, Session Buddy (quietly) changed the frequency of it's updates:

https://groups.google.com/forum/#!topic/sessionbuddy-discuss/QUb4zqWz5CI

So I was totally screwed.  In light of this recent little disaster, I did a some searching and experimenting with different extensions and now I'm using this:

https://chrome.google.com/webstore/detail/tabs-backup-restore/dehocbglhkaogiljpihicakmlockmlgd

So far good. Try it... Seems better, right? So, anyway, I recommend it guys. 

(btw, I'm not putting down @deanoemcke , it looks like it was inevitable. (I'm a developer too.)  )

**Comment by SnIPeRSnIPeR** [comment]
**Tabs Backup & Restore**
updated: 14 november 2012...

**Comment by RobertAb1** [comment]
Tabs Backup & Restore not maintained anymore.

**Comment by mkrier** [comment]
Oh I see.  Ok what's better then?  In the void left of Session Buddy?

**Comment by RobertAb1** [comment]
We have problem here with Chrome. 

Maybe Firefox 61 + Session Sync (or MySessions)? :). 
Or Firefox 56/Waterfox 56.2.2 + Session Manager (works even better).
Firefox has already lazy tabs implemented natively. 

But seriously, how often do you restart your Chrome? I have noticed on my that Session Buddy is saving session 1 time every 1 restart. And after problem caused by The Great Suspender, I was able to recover my tabs using Session Buddy's saved session.

**Comment by mkrier** [comment]
You're in the minority there. As I said in the op post- session buddy quietly stretched out the gap for it's automatic saves.  hell, does it even do automatic saves at all anymore?  

I was NOT able to recover my tabs.  @deanoemcke  can maybe chime in about it from the other post thread.

And no, it's infeasible for me to  switch to firefox.

**Comment by RobertAb1** [comment]
How often do you restart your Chrome?

**Comment by SnIPeRSnIPeR** [comment]
For me, it's "Session Buddy":

![2018-08-28_21-53-58](https://user-images.githubusercontent.com/31162059/44731254-fac56d80-ab0c-11e8-9158-be1e4d6519e8.png)

**Comment by mkrier** [comment]
@RobertAb1  

mmm well it's hard to take an average.  The mean wouldn't be a good measure, but the median would be like... call it about two weeks...

**Comment by mkrier** [comment]
@RobertAb1 

Well, we philosophically disagree about backups.  I believe that backups should never be manual.  Humans are usually the point of failure.  I shouldn't have to think about backups at all, except-in-so-much as to check that they're actually still working.  (see- even there, that was the point of failure; the human.)

I'm much happier with Tabs Backup & Restore even if it's not supported.

**Comment by RobertAb1** [comment]
@mkrier

I completely agree: backups should be done automatically. And they are during restart :)

But I need to tell you. Session Manager in Firefox/Waterfox never/ever gave me problems. Never. The best session manager ever made. Also in Firefox/Waterfox, I never have problems with the order of tabs in window even after some accidents (like Windows blue screen). In Chrome, the order of tabs is ALWAYS changed. That is way, I am using now Waterfox.

**Comment by mkrier** [comment]
@RobertAb1  

Ok I respect that, but manually restarting it to cause a backup, is still.... manual.  

And while Session Buddy never gave you, in particular, problems, well- it gave a lot of other people problems who didn't waste their human labor doing things manually.  (as you'll see in the link I provided in the op post- people who got screwed over by the quiet update last winter).

And, again, switching to firefox (waterfox), isn't an option for me- I rely on other extensions here.  (One in particular)

But anyway, I think we've got our different situations resolved, right. 

If so, cheers,

m

**Comment by RobertAb1** [comment]
@mkrier
At least, I will know not to worry about Session Buddy.
The main reason, I am restarting Chrome is to speed it up, and to make it to use less resources.
Have a nice day.

**Comment by mkrier** [comment]
@RobertAb1  oohhh I thought you were restarting it specifically to make session buddy backup. I see now, thanks.

**Comment by deanoemcke** [comment]
My 2 cents. I have experienced the same issue as @mkrier.

I think we just need to define what we are expecting from a session manager. For me, I expect it to be constantly recording the state of my tabs. If chrome crashes, I want it to have remembered my last known state. 

- Session Buddy used to do this, but now no longer does. This for me makes it useless for my primary use case.
- The Great Suspender _does_ do this. It will update the most recent session whenever a url is changed or a tab is added/removed. However, we're talking about an external tool for when TGS fails so that's out in this instance.
- Tabs Backup & Restore seems to do exactly what I want. It's a shame it's no longer maintained, but it still does exactly what I need, so personally i've switched to using this.

**Comment by ahmedhamdy90** [comment]
I have switched too to Tabs Backup and Restore, at least it is working for now, until we see what shall happen in the next Chrome update...

**Comment by mikhoul** [comment]
[**@RobertAb1**](https://github.com/RobertAb1) commented on [28 août 2018 à 06:01 UTC−4](https://github.com/deanoemcke/thegreatsuspender/issues/742#issuecomment-416525422 "2018-08-28T10:01:30Z - Replied by Github Reply Comments"):
> Tabs Backup &amp; Restore is not maintained anymore.

But does it work fine ?  🤔

If it is not maintained but is working fine there is no real issue.


Also I use this extension here: https://chrome.google.com/webstore/detail/freshstart-cross-browser/nmidkjogcjnnlfimjcedenagjfacpobb

There is  a crash recovery integrated but lately it don,t seem to save as it should my session, maybe it's a conflict withanother extension I have since before it was working fine and saved my but lot of times 😊.

**Comment by chrcoluk** [comment]
Agreed, just because its not maintained it doesnt mean it cannot be used, not maintained can be a good thing as it means no new bugs been introduced from broken new code.  As well as feature trimming which session buddy has suffered from.

**Comment by liamjohnston** [comment]
i'll just leave this here
![Uploading Screen Shot 2018-10-04 at 11.04.45 AM.png…]()

**Comment by Cerberus-tm** [comment]
@RobertAb1 So for you Tab Session Manager works well in Firefox? It has a score 2.9 stars and I see a few reviews complaining about how it isn't reliable... (I'm still preparing for my eventual switch to the new Firefox).

**Comment by Luckz** [comment]
Maybe I'm in the minority here, but to me a "session manager" needs to maintain scroll position for all tabs, back/forwards history for all tabs, and perhaps have the option to keep unsent formdata. Saving a tab (versus piecing together a tab from whatever you exported) was only possible in Firefox before Web Extensions.
Tabs affected by The Great Suspender already lose scroll position much more than a native (merely discarded) tab would, cannot keep forward history because it is overwritten by doing a web navigation, and TGS pollutes back history with duplicates of the current tab (one for every time you suspend).

Chrome session managers can by design only be bookmark tools.

----

To add to the links mentioned in this thread, 
https://chrome.google.com/webstore/detail/tabs-outliner/eggkanocgddhmamlbiijnphhppkpkmkl?hl=en and https://chrome.google.com/webstore/detail/tabfern-tab-manager-and-b/hbajjpcdbninabigakflkhiogmmjaakm?hl=en add neat vertical own-window views.

**Comment by CollinChaffin** [comment]
I've always run SB and even with current version, mine has no issue on MANY systems not only instantly saving current session (open Chrome, open 5 tabs, close Chrome 10 sec later, reopen fresh, and there is the auto-saved session to restore as well as the HUNDREDS of sessions spanning years - all of which restore fine).

All I can think is it's something with your settings, or a conflicting extension.  Also, perhaps your SB database has an issue?

**Comment by ckyoog** [comment]
[Tab Session Manager](https://github.com/sienori/Tab-Session-Manager) is excellent, IMO, with features like auto save, auto cloud sync, and most important, it is open source.
